src/wallet/test/wallet_tests.cpp

```diff
@@ -631,18 +630,17 @@ BOOST_FIXTURE_TEST_CASE(GetLegacyBalance_coinbase_maturity, TestChain100Setup) {
     CreateAndProcessBlock({}, GetScriptForDestination(WitnessV0KeyHash(coinbaseKey.GetPubKey().GetID())));
   }
 
-  // As per mature outputs we should have 103 blocks worth of rewards
-  // - 1 reward used to stake the watch-only + the initial stake + the
-  // watch-only stake and reward
+  // As per mature outputs we should have 103 blocks worth of rewards + the initial
+  // stake + the watch-only reward
   {
       auto coinbase_reward = m_coinbase_txns.back().vout[0].nValue;
       LOCK2(cs_main, m_wallet->cs_wallet);
       const CAmount all_balance = m_wallet->GetLegacyBalance(ISMINE_ALL, 0, nullptr);
       const CAmount spendable_balance = m_wallet->GetLegacyBalance(ISMINE_SPENDABLE, 0, nullptr);
       const CAmount watchonly_balance = m_wallet->GetLegacyBalance(ISMINE_WATCH_ONLY, 0, nullptr);
-      BOOST_CHECK_EQUAL(all_balance, (10000 * UNIT) + coinbase_reward * 102 + watch_only_coinbase->GetValueOut());
-      BOOST_CHECK_EQUAL(spendable_balance, (10000 * UNIT) + coinbase_reward * 102);
-      BOOST_CHECK_EQUAL(watchonly_balance, watch_only_coinbase->GetValueOut());
+      BOOST_CHECK_EQUAL(all_balance, (10000 * UNIT) + coinbase_reward * 103 + watch_only_coinbase->vout[0].nValue);
+      BOOST_CHECK_EQUAL(spendable_balance, (10000 * UNIT) + coinbase_reward * 103);
+      BOOST_CHECK_EQUAL(watchonly_balance, watch_only_coinbase->vout[0].nValue);
```

Comment:
Nit: Would maybe make sense to introduce a getter like `GetReward` or smth like that

Reply:
I'd rather not add a getter just to use in tests, I will create a `GetReward()` function in the test, WDYT?

Reply:
Are there not more call sites which access the reward (validation for example) which are not just in tests?

This does not have to happen in this pull request, but form a data encapsulation standpoint I'd welcome that very much.

Reply:
There actually are (not with the `->` operator, but from a normal reference). There are quite a few of those usages, so I agree it does make sense. I'll follow up in an another PR.

Reply:
Maybe it would be better to do it after finalization rewards are merged.